crates/sozo/src/ops/events.rs

```diff
@@ -29,7 +38,120 @@ pub async fn execute(args: EventsArgs, env_metadata: Option<Environment>) -> Res
 
     let res = provider.get_events(event_filter, continuation_token, chunk_size).await?;
 
-    let value = serde_json::to_value(res)?;
-    println!("{}", serde_json::to_string_pretty(&value)?);
+    if json {
+        let value = serde_json::to_value(res)?;
+        println!("{}", serde_json::to_string_pretty(&value)?);
+    } else {
+        parse_and_print_events(res, events_map);
+    }
     Ok(())
 }
+
+fn parse_and_print_events(
+    res: starknet::core::types::EventsPage,
+    events_map: HashMap<String, Vec<Event>>,
+) {
+    println!("Continuation token: {:?}", res.continuation_token);
+    println!("----------------------------------------------");
+    for event in res.events {
+        if let Some(e) = parse_event(event.clone(), &events_map) {
+            println!("{}\n", e);
+        } else {
+            // Couldn't parse event
+            println!("{}\n", serde_json::to_string_pretty(&event).unwrap());
+        }
+    }
+}
+
+fn parse_event(
+    event: starknet::core::types::EmittedEvent,
+    events_map: &HashMap<String, Vec<Event>>,
+) -> Option<String> {
+    let keys = event.keys;
+    let event_hash = keys[0].to_string();
+    let Some(events) = events_map.get(&event_hash) else { return None };
+
+    'outer: for e in events {
+        let mut ret = format!("Event name: {}\n", e.name);
+        let mut data = VecDeque::from(event.data.clone());
+
+        // Length is two only when its custom event
+        if keys.len() == 2 {
+            let name = parse_cairo_short_string(&keys[1]).ok()?;
+            ret.push_str(&format!("Component name: {}", name));
+        }
+
+        match &e.kind {
+            EventKind::Struct { members } => {
+                for field in members {
+                    if field.kind != EventFieldKind::DataSerde {
+                        continue;
+                    }
+                    match field.ty.as_str() {
```

Comment:
would be nice if we could implement `Display` on `field.ty` here

Reply:
this is basically parsing raw felts into human readable formats? that is done in various other places in the codebase and more in the future too, for example, when using `sozo execute`. so seems it could be useful as a reusable piece of code. basically, the flow of `take raw rpc value` -> `use manifest to parse` -> `display human readable version` will be useful for other cli commands too